sympy/functions/elementary/trigonometric.py

```diff
@@ -2251,6 +2251,16 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_is_real(self):
+        x = self.args[0]
+        if x.is_real == False:
+            return False
+        y = abs(x) - 1
```

Comment:
Actually the user must not see any type error unless he passed in an improper type (in which case the function's docstring must mention which types it will accept).
At least that's what my book says, but I tend to agree with it unless very special circumstances override it.